### DISCUSSION TITLE
--renew-hook that does not fire even if cert updated.

### DISCUSSION THREAD

**Discussion by mrzaz** [acmesh-official/acme.sh#6172] [Q&A]
Hello,
I am getting a bit gray haired due to an issue I can't understand.

When I execute the -issue command as follows: (yes, I am right now forcing it to update)

mydomain.com is not the real domain used. Just a bogus one for this example.

./acme.sh --issue --force --ecc --output-insecure --log-level 2 --log output.log --pre-hook "echo pre-hook" --post-hook "echo post-hook" **--renew-hook "echo > $homedir/domain_renewed.txt; echo renew-hook executed"** --dns dns_dynu --server letsencrypt --accountconf account1_mydomain.com.conf -d *.mydomain.com

It fires "--pre-hook" and "--post-hook" as can be seen in logs.
BUT it **never** fires "--renew-hook" even if the certificate is getting renewed.

I even checked with "debug=3" and could only find the following reference

[Wed Dec 25 19:38:43 CET 2024] ./acme.sh:_setopt:2351                   4:**Le_PreHook**='__ACME_BASE64__START_ZWNobyBwcmUtaG9vaw==__ACME_BASE64__END_'
[Wed Dec 25 19:38:44 CET 2024] ./acme.sh:_base64:982                    base64 single line.
[Wed Dec 25 19:38:44 CET 2024] ./acme.sh:_setopt:2327                   OK
[Wed Dec 25 19:38:44 CET 2024] ./acme.sh:_setopt:2351                   5:**Le_PostHook**='__ACME_BASE64__START_ZWNobyBwb3N0LWhvb2s=__ACME_BASE64__END_'
[Wed Dec 25 19:38:44 CET 2024] ./acme.sh:_base64:982                    base64 single line.
[Wed Dec 25 19:38:44 CET 2024] ./acme.sh:_setopt:2327                   OK
[Wed Dec 25 19:38:45 CET 2024] ./acme.sh:_setopt:2351                   6:**Le_RenewHook**='__ACME_BASE64__START_dG91Y2ggL2hvbWUvQWRtaW5pc3RyYXRvci8uYWNtZS5zaC9kb21haW5fcmVuZXdlZC50eHQ7IGVjaG8gcmVuZXctaG9vayBleGVjdXRlZA==__ACME_BASE64__END_'
[Wed Dec 25 19:38:45 CET 2024] ./acme.sh:_setopt:2327                   OK

[Wed Dec 25 19:38:46 CET 2024] Runing pre hook:'echo pre-hook'
pre-hook

[Wed Dec 25 19:26:34 CET 2024] Running post hook:'echo post-hook'
post-hook

Below is when catching exit code in bash script from .scme script
Exitcode: 0 , Certificate request for mydomain.com successful.

Seen in log
[Wed Dec 25 19:39:25 CET 2024] Cert success.
.
.
-----BEGIN CERTIFICATE-----
.
.
[Wed Dec 25 19:39:25 CET 2024] Your cert is in: /home/Administrator/.acme.sh/*.mydomain.com_ecc/*.mydomain.com.cer
[Wed Dec 25 19:39:25 CET 2024] Your cert key is in: /home/Administrator/.acme.sh/*.mydomain.com_ecc/*.mydomain.com.key
[Wed Dec 25 19:39:26 CET 2024] The intermediate CA cert is in: /home/Administrator/.acme.sh/*.mydomain.com_ecc/ca.cer
[Wed Dec 25 19:39:26 CET 2024] And the full-chain cert is in: /home/Administrator/.acme.sh/*.mydomain.com_ecc/fullchain.cer
.
.
[Wed Dec 25 19:39:26 CET 2024] ./acme.sh:_setopt:2351                   12:Le_CertCreateTime='1735151966'
[Wed Dec 25 19:39:26 CET 2024] ./acme.sh:_setopt:2327                   OK
[Wed Dec 25 19:39:27 CET 2024] ./acme.sh:_setopt:2351                   13:Le_CertCreateTimeStr='2024-12-25T18:39:26Z'
[Wed Dec 25 19:39:27 CET 2024] ./acme.sh:_setopt:2327                   OK
[Wed Dec 25 19:39:27 CET 2024] ./acme.sh:_setopt:2351                   26:Le_RenewalDays='60'
[Wed Dec 25 19:39:28 CET 2024] ./acme.sh:_setopt:2327                   OK
[Wed Dec 25 19:39:28 CET 2024] ./acme.sh:_setopt:2351                   14:Le_NextRenewTimeStr='2025-02-22T18:39:26Z'
[Wed Dec 25 19:39:28 CET 2024] ./acme.sh:_setopt:2327                   OK
[Wed Dec 25 19:39:28 CET 2024] ./acme.sh:_setopt:2351                   15:Le_NextRenewTime='1740249566'

Could someone please explain why the --renew-hook does not fire even if cert is renewed ?

I am always running latest version of .acme.sh 

//Dan Lundqvist

**Comment by neilpang** [comment]
Expected behavior. `--renew-hook` fires only on the renewal path (`--renew` / cron) -- gated on `_ACME_IS_RENEW`, which `--issue` never sets, even with `--force`. Your log shows `Le_RenewHook` was saved, so it will fire on future cron renewals.

Test it with: `acme.sh --renew -d '*.mydomain.com' --ecc --force`.

For a command on every issuance *and* renewal, use `--post-hook` instead.
